url85e313f409_edugasser_Ascensor_tgz-pJ8-ascensor_MainJ8/src/ascensor/Main.java

```diff
@@ -0,0 +1,778 @@
+         if (actual < destino)
+         {
+             ascensor.setDireccion("ASC");
+         }else{
+             ascensor.setDireccion("DESC");
+         }
+     }
+     public void salida_ascensor()
+     {
+//System.out.println("SALIDA_ASCENSOR: " + clock);
+        if (ascensor.getNumPasajeros() < MAX_PASAJEROS) aceptar_pasajeros();
+        piso_destino = proxima_parada();
+        if (piso_destino == piso_actual ){
+            event_list.setS((clock + 3));
+        }else{
+            ascensor.setViajando(true);  
+            event_list.setL(clock + (Tviaje*(Math.abs(piso_destino - piso_actual))) + Tcerrar + Tabrir);
+            event_list.setS(INFINITO);
+           //System.out.println("salida ascensor: llegada ascensor: " + event_list.getL() + " piso_destino: " + piso_destino + " piso_Actual: " + piso_actual+ " "+ Tcerrar +" "+ Tabrir);
+   
+        }       
+    }
+     
+     public int sig_piso(int p, int b[],boolean t)
+     {
+         int i = p;
+         /* si hay llamadas internas en el mismo piso, no debe mirar si hay llamadas externas en ese mismo piso
+          * por que puede que haya alguna llamada porque ya no entran más pasajeros
+          */
+         if (t) i = p+1;
+         boolean salir = false;
+         while (i < MAX_PISOS && !salir){
+             if (b[i] == 1) salir = true;
+             i++;
+         }
+         if (salir){
+             return --i;
+         }else{
+             return -1;//no hay llamadas en los pisos siguientes
+         }
+     }
+     public boolean hay_sig_piso(int p, int b[])
+     {
+         int i = p;
+         boolean salir = false;
+         
+         while (i < MAX_PISOS && !salir){
+             if (b[i] == 1) salir = true;
+             i++;
+         }
+         return salir;
+     }
+     public boolean hay_ant_piso(int p, int b[])
+     {
+         int i = p;
+         boolean salir = false;
+         while (i >= 0 && !salir){
+              
+             if (b[i] == 1) salir = true;
+             --i;
+         }
+         return salir;
+     }
+     public int ant_piso(int p, int b[], boolean t)
+     {
+         int i = p;
+         if (t) i = p-1;
+         
+         boolean salir = false;
+         while (i >= 0 && !salir){
+              
+             if (b[i] == 1) salir = true;
+             --i;
+         }
+         if (salir){
+             return ++i;
+         }else{
+             return -1;
+         }
+     }
+
+     public boolean hay_llamadas_internas(){
+         int i = 0;
+         boolean salir = false;
+         while ( i < MAX_PISOS && !salir)
+         { 
+             if (internas[i] == 1){
+                 salir = true;
+             }
+             i++;
+         }
+         if (salir){
+             return true;
+         }else{
+             return false;
+         }
+     }
+     /* obtenemos la siguiente parada dependiendo si hay peticiones en otras plantas */
+     public int proxima_parada()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.